### ISSUE TITLE
Do not log info messages as errors

### ISSUE THREAD

**Issue by akkie** [vorburger/MariaDB4j#27]
The lib logs a lot of messages as error messages. I think this messages should be more considered as info messages.

```
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:12 [Note] /tmp/MariaDB4j/base/bin/mysqld: Normal shutdown
[error] c.v.e.ManagedProcess - mysqld: 
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:12  InnoDB: Starting shutdown...
```

```
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:14 InnoDB: The InnoDB memory heap is disabled
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:14 InnoDB: Mutexes and rw_locks use GCC atomic builtins
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:14 InnoDB: Compressed tables use zlib 1.2.3
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:14 InnoDB: Using Linux native AIO
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:14 InnoDB: Initializing buffer pool, size = 128.0M
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:14 InnoDB: Completed initialization of buffer pool
[error] c.v.e.ManagedProcess - mysqld: InnoDB: The first specified data file ./ibdata1 did not exist:
[error] c.v.e.ManagedProcess - mysqld: InnoDB: a new database to be created!
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:14  InnoDB: Setting file ./ibdata1 size to 10 MB
[error] c.v.e.ManagedProcess - mysqld: InnoDB: Database physically writes the file full: wait...
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:14  InnoDB: Log file ./ib_logfile0 did not exist: new to be created
[error] c.v.e.ManagedProcess - mysqld: InnoDB: Setting log file ./ib_logfile0 size to 5 MB
[error] c.v.e.ManagedProcess - mysqld: InnoDB: Database physically writes the file full: wait...
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:14  InnoDB: Log file ./ib_logfile1 did not exist: new to be created
[error] c.v.e.ManagedProcess - mysqld: InnoDB: Setting log file ./ib_logfile1 size to 5 MB
[error] c.v.e.ManagedProcess - mysqld: InnoDB: Database physically writes the file full: wait...
[error] c.v.e.ManagedProcess - mysqld: InnoDB: Doublewrite buffer not found: creating new
[error] c.v.e.ManagedProcess - mysqld: InnoDB: Doublewrite buffer created
[error] c.v.e.ManagedProcess - mysqld: InnoDB: 127 rollback segment(s) active.
[error] c.v.e.ManagedProcess - mysqld: InnoDB: Creating foreign key constraint system tables
[error] c.v.e.ManagedProcess - mysqld: InnoDB: Foreign key constraint system tables created
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:14  InnoDB: Waiting for the background threads to start
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:15 Percona XtraDB (http://www.percona.com) 5.5.33a-MariaDB-31.1 started; log sequence number 0
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:15 [Note] Plugin 'FEEDBACK' is disabled.
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:15 [Note] Server socket created on IP: '0.0.0.0'.
[error] c.v.e.ManagedProcess - mysqld: 150730  9:41:15 [Note] /tmp/MariaDB4j/base/bin/mysqld: ready for connections.
[error] c.v.e.ManagedProcess - mysqld: Version: '5.5.33a-MariaDB'  socket: '/tmp/MariaDB4j.3333.sock'  port: 3333  MariaDB Server
```

**Comment by vorburger** [comment]
I'm not sure how easy this is to fix.. These are the logs from the
externally launched mysqld proc. If you have the time to investigate the
root cause underlying this, that would be very helpful. Have I perhaps
confused a STDOUT vs. STDERR somewhere??

**Comment by kbjp2** [comment]
Is there any work-around for this problem ? ...Or even better, has this been fixed already ?

**Comment by vorburger** [comment]
@akkie and @kbjp2 I've just looked at this again for a moment, and done a quick test:

    cd /tmp/MariaDB4j/base
    rm -rf /tmp/MariaDB4j/data/1
    /tmp/MariaDB4j/base/bin/mysql_install_db --datadir=/tmp/MariaDB4j/data/1 --basedir=/tmp/MariaDB4j/base --no-defaults --force --skip-name-resolve >stdout.txt 2>stderr.txt

Doing this, we can see in stderr.txt that mariadb's mysql/mysqld/mysql_install_db really do write all those whole lot of "[Note]" messages out to STDERR, and only some messages to STDOUT.

MariaDB4j channels everything written by an external process into STDERR into an slf4j logger with severity error, and everything into STDOUT into an info log. So it's ... doing the right thing, technically.

As to why mariadb writes all these clearly more informational than error types messages out to STDERR, I'm afraid I simply don't know. Perhaps you could raise this somewhere like on Stack Overflow or a mariadb Forum or whevever. If there are any CLI options to make mariadb log differently, we could pass them in MariaDB4j.

**Comment by hanklank** [comment]
I couldn't find a quick solution, but I reported to the MariaDB jira. https://jira.mariadb.org/browse/MDEV-10451

**Comment by vorburger** [comment]
https://jira.mariadb.org/browse/MDEV-10451 mentions new start-up options --log-warnings and/or --silent-startup .. if anyone on this wants to play with those and put in a pull request for something based on that, I'm all for it.

The point of "filter the output in your library to show only messages with the [ERROR] tag" is something I didn't even think of - pragmatic, why not. We could that as well - it would certainly be possible to "fix up" things on our end (so e.g. we would send stuff gone into STDERR without [ERROR] tag into the slf4j info logger after all). If anyone wants to have a go in this direction, certainly do feel free as well. (But if you do, please make the implementation "clean" in the code, as in don't completely hard-code this in the ch.vorburger.exec package, but make it pluggable there and override this specific mariadb thing somewhere in the ch.vorburger.mariadb4j package.)

**Comment by vorburger** [comment]
@hanklank @kbjp2 @akkie in https://github.com/vorburger/MariaDB4j/compare/issue27_infoVsErrorLogging (not on master, yet) I've just implemented a proposed "fix" for this issue (which is more of a "heuristic workaround" really), based on the idea to filter the output to send messages sent to STDERR containing 'error' or 'ERROR' to LOG.error and others to LOG.info ... perhaps you'd like to test it out for yourselves and provide some feedback if this functionally does the trick for you, and if you have any technical / code review feedback? I'd like to get this into a 2.2.2 release within 1 week or so. Thanks for your support to MariaDB4j!

**Comment by vorburger** [comment]
Fix merged as ba63bcab142ebfd2392a62cc105100198b3efecb; done.
